sqlmesh/core/loader.py

```diff
@@ -58,23 +58,17 @@ def update_model_schemas(
         if not model:
             continue
 
-        default = model_defaults.get(name)
-        kwargs = {
-            "default_schema": default.schema_ if default else None,
-            "default_catalog": default.catalog if default else None,
-        }
-        table = set_default_schema_and_catalog(name, **kwargs)
         try:
-            model.update_schema(schema, **kwargs)
+            model.update_schema(schema, default_catalog=default_catalog)
             optimized_query_cache.with_optimized_query(model)
 
             columns_to_types = model.columns_to_types
             if columns_to_types is not None:
-                schema.add_table(table, columns_to_types, dialect=model.dialect)
+                schema.add_table(model.fqn, columns_to_types, dialect=model.dialect)
         except SchemaError as e:
             if "nesting level:" in str(e):
                 logger.error(
-                    "SQLMesh requires all model names and references to have the same level of nesting. You can set schema and catalog in model_defaults."
+                    "SQLMesh requires all model names and references to have the same level of nesting."
```

Comment:
people will still run into this in airflow right? what's the recourse for those folks

Reply:
This is the regression that I mentioned during standup. Until Airflow is implemented there is no recourse for those folks.